builds/jenkins-build-pack-sls-app/Jenkinsfile

```diff
@@ -0,0 +1,942 @@
+              def arn = name2ArnPair[1].trim()
+              if(arn.startsWith('arn')) {
+                 def arnAsArray = arn.split(':') // Here we splitting the arn itself in hope to obtain type of the resource. Should be the third element: arn:aws:dynamodb:us-east-1:102707241671:table/jazzUsersTable53
+                 if(arnAsArray.size() > 2) { // Making sure that the third element exists
+                   def artifactType = arnAsArray[2]
+                   if(artifactType == 'kinesis') artifactType = 'kinesis_stream' // 'kinesis_stream' needs to be instaead of simple 'kinesis' that did not get persisted :-( by itself
+                   events.sendCompletedEvent('CREATE_ASSET',
+                                             null,
+                                             utilModule.generateAssetMap("aws",
+                                                                         arn,
+                                                                         artifactType,
+                                                                         config),
+                                             environment_logical_id)
+                   echo "Sending type $artifactType with arn $arn"
+
+                }
+              }
+            }
+          }
+
+          if (domain != "jazz") {
+            createSubscriptionFilters(config, accountDetails, env_key, credsId);
+          }
+					// reset Credentials
+          resetCredentials(credsId)
+
+          def svc_response = echoServiceInfo(environment_logical_id)
+          send_status_email(config, 'COMPLETED', svc_response)
+
+        } catch (ex) {
+          send_status_email(config, 'FAILED', '')
+          events.sendFailureEvent('UPDATE_ENVIRONMENT', ex.getMessage(), environmentDeploymentMetadata.generateEnvironmentMap("deployment_failed", environment_logical_id, null), environment_logical_id)
+          events.sendFailureEvent('UPDATE_DEPLOYMENT', ex.getMessage(), environmentDeploymentMetadata.generateDeploymentMap("failed", environment_logical_id, gitCommitHash), environment_logical_id)
+          events.sendFailureEvent('DEPLOY_TO_AWS', ex.getMessage(), context_map, environment_logical_id)
+          error ex.getMessage()
+        }
+
+        def serviceContext = [created_by: config['created_by'], deployed_by: gitCommitOwner]
+
+        events.sendCompletedEvent('UPDATE_ENVIRONMENT', 'Environment update event for deployment completion', environmentDeploymentMetadata.generateEnvironmentMap("deployment_completed", environment_logical_id, null), environment_logical_id)
+        events.sendCompletedEvent('UPDATE_DEPLOYMENT', "Deployment completion Event for ${env_key} deployment", environmentDeploymentMetadata.generateDeploymentMap("successful", environment_logical_id, gitCommitHash), environment_logical_id)
+        events.sendCompletedEvent('DEPLOY_TO_AWS', 'Successfully deployed services to AWS', serviceContext, environment_logical_id)
+
```

Comment:
lambda functions's permissions need to be updated so that primary account's platform services role can invoke them. its used for test function capability

Reply:
Skipping this for now, since sls-app does not have test lambda funtionality